server/graphql/v2/input/AccountInput.js

```diff
@@ -0,0 +1,35 @@
+import { GraphQLInt, GraphQLString, GraphQLInputObjectType } from 'graphql';
+
+import models from '../../../models';
+import { idDecode } from '../identifiers';
+import { NotFound } from '../../errors';
+
+const AccountInput = new GraphQLInputObjectType({
+  name: 'AccountInput',
+  fields: () => ({
+    id: { type: GraphQLString },
+    slug: { type: GraphQLString },
+    legacyId: { type: GraphQLInt },
+  }),
+});
+
+const fetchAccountWithInput = async input => {
+  let collective;
+  if (input.id) {
+    const id = idDecode(input.id, 'account');
+    collective = await models.Collective.findByPk(id);
+  } else if (input.legacyId) {
+    collective = await models.Collective.findByPk(input.legacyId);
+  } else if (input.slug) {
+    const slug = input.slug.toLowerCase();
+    collective = await models.Collective.findBySlug(slug);
+  } else {
+    return new Error('Please provide a slug or an id');
```

Comment:
Shouldn't it be throw rather than return?

```suggestion
    throw new Error('Please provide a slug or an id');
```